### PR TITLE
Add `model.eval()` in TensorBoad graph visualization to avoid BN stats changes

### PR DESCRIPTION
**Pull request by Y-T-G**
The `trace()` call here causes BN stats to change since the `trainer.model` is not in eval mode:
https://github.com/ultralytics/ultralytics/blob/e138d701a0df99eed43c83b7a63671a5c1c43656/ultralytics/utils/callbacks/tensorboard.py#L48

Fix by putting `trainer.model` to `eval` mode before trace.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced model visualization support in TensorBoard.

### 📊 Key Changes
- Added an evaluation mode setting for the model before graph tracing in TensorBoard logging.

### 🎯 Purpose & Impact
- **Purpose**: To ensure that the model's forward pass is in the evaluation setting, promoting more accurate graph representations within TensorBoard.
- **Impact**: Developers and users can get a clearer visualization of the model structure during inference, potentially aiding in debugging and understanding the model's behavior. This could also lead to fewer errors when generating the graph under different operation modes. 📈🔍

### CODE REVIEW

**Comment by Y-T-G** [comment]
I have read the CLA Document and I sign the CLA

**Comment by Y-T-G** [comment]
recheck

**Comment by glenn-jocher** [comment]
@Y-T-G absolutely, let's recheck that for you! If you encounter any further issues or have questions, don't hesitate to reach out. Happy coding! 😊

**Comment by glenn-jocher** [comment]
@Y-T-G thanks for the PR, but not sure if this is wise as we'll actually be changing the underlying model. If it changes mode (from .train() to .eval()) then won't that affect training later after this callback is complete?

**Comment by Y-T-G** [comment]
> @Y-T-G thanks for the PR, but not sure if this is wise as we'll actually be changing the underlying model. If it changes mode (from .train() to .eval()) then won't that affect training later after this callback is complete?

Before the training epoch begins, it calls model.train(), so it shouldn't be an issue:
https://github.com/ultralytics/ultralytics/blob/9c4259614523cc72b5471b6a0677746e8be39a4e/ultralytics/engine/trainer.py#L349

**Comment by Y-T-G** [comment]
Currently, it affects the model because it is changing the BN stats even before the training begins which shouldn't happen.

**Comment by glenn-jocher** [comment]
@Y-T-G ok got it. Let's give this a try then! Merging PR. Please let me know if this correctly resolves the modified BN stats for you.
